libarchive/archive_util.c

```diff
@@ -77,7 +77,7 @@
 #define O_CLOEXEC	0
 #endif
 
-static int archive_utility_string_sort_helper(char **, unsigned int);
+static int archive_utility_string_sort_helper(const void *, const void *);
```

Comment:
Just like the callback for `bsearch` in `archive_pack_dev.c`, this may need to be decorated with `__LA_LIBC_CC`:

Otherwise, it may fail to compile in some configurations on Windows x86.

https://github.com/libarchive/libarchive/blob/59b09796c3402489c3f47625b6401a28d24ad8a4/libarchive/archive_pack_dev.c#L73

https://github.com/libarchive/libarchive/blob/59b09796c3402489c3f47625b6401a28d24ad8a4/libarchive/archive_pack_dev.c#L305-L307

See #1623 for more details :)